ledger_device_sdk/src/io.rs

```diff
@@ -268,7 +280,80 @@ impl Comm {
         }
     }
 
-    pub fn decode_event<T>(&mut self, spi_buffer: &mut [u8; 128]) -> Option<Event<T>>
+    pub fn next_event_ahead<T>(&mut self) -> bool
+    where
+        T: TryFrom<ApduHeader>,
+        Reply: From<<T as TryFrom<ApduHeader>>::Error>,
+    {
+        // if self.event_pending {
+        //     let event: Option<Event<T>> = self.check_event();
+        //     if event.is_some() {
+        //         return true;
+        //     }
+        // }
+
+        let mut spi_buffer = [0u8; 128];
+
+        // unsafe {
+        //     G_io_app.apdu_state = APDU_IDLE;
```

Comment:
Calling this each time we enter the `next_event_ahead` function messes up the IO global state and results in the reply not being sent for instance after signing a transaction.

Maybe we should call it once in an init function before looping on `next_event_ahead` ? @kingofpayne

Reply:
I think the `next_event_ahead` method should not call again `seph_recv` when an APDU is pending. When an APDU is pending, the MCU expects a reply to that event, but if you call again `next_event_ahead` while the response has not been sent I think this is where you get troubles (that could explain why you had to comment out the global state init).